component/block_cache/block_cache.go

```diff
@@ -80,24 +84,26 @@ type BlockCache struct {
 
 // Structure defining your config parameters
 type BlockCacheOptions struct {
-	BlockSize      uint64 `config:"block-size-mb" yaml:"block-size-mb,omitempty"`
-	MemSize        uint64 `config:"mem-size-mb" yaml:"mem-size-mb,omitempty"`
-	TmpPath        string `config:"path" yaml:"path,omitempty"`
-	DiskSize       uint64 `config:"disk-size-mb" yaml:"disk-size-mb,omitempty"`
-	DiskTimeout    uint32 `config:"disk-timeout-sec" yaml:"timeout-sec,omitempty"`
-	PrefetchCount  uint32 `config:"prefetch" yaml:"prefetch,omitempty"`
-	Workers        uint32 `config:"parallelism" yaml:"parallelism,omitempty"`
-	PrefetchOnOpen bool   `config:"prefetch-on-open" yaml:"prefetch-on-open,omitempty"`
+	BlockSize      float64 `config:"block-size-mb" yaml:"block-size-mb,omitempty"`
```

Comment:
Why are we using floating values for BlockSize?

Reply:
We take input in terms of MBs. In initial reading block-cache there were couple of requests to make it in bytes instead as customer wanted to read only few bytes from a very large file and there no need to even download 1 mb data. Making this float gives us an oppurtunity to specify it in bytes. For e.g. a value "0.5" will mean 512KB bufer and "0.05" or values like these can take it further down.